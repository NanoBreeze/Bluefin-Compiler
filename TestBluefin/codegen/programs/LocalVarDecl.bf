/*
Verifies local variables are declared (on the stack)
*/

void foo(int a) {
	int b;	 // undefined value
	int c = 5; // primary int
	bool d = true;
	float e = 6.0;

	int f = 7+a*2;
	float g = 5-a; // casting '5-a'
	bool h = 5>a; 
}