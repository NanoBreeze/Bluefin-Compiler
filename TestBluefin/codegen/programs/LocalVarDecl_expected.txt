; ModuleID = 'LocalVarDecl.bf'
source_filename = "LocalVarDecl.bf"

define void @foo(i32 %a) {
entry:
  %b = alloca i32, align 4
  %c = alloca i32, align 4
  store i32 5, i32* %c, align 4
  %d = alloca i1, align 1
  store i1 true, i1* %d, align 1
  %e = alloca float, align 4
  store float 6.000000e+00, float* %e, align 4
  %f = alloca i32, align 4
  %multmp = mul i32 %a, 2
  %addtmp = add i32 7, %multmp
  store i32 %addtmp, i32* %f, align 4
  %g = alloca float, align 4
  %subtmp = sub i32 5, %a
  %casttmp = sitofp i32 %subtmp to float
  store float %casttmp, float* %g, align 4
  %h = alloca i1, align 1
  %cmpSGTtmp = icmp sgt i32 5, %a
  store i1 %cmpSGTtmp, i1* %h, align 1
  ret void
}
